assets/wizards/popups/views/segments/SegmentsList.js

```diff
@@ -131,6 +132,14 @@ const SegmentActionCard = ( {
 		resortSegments( target );
 	};
 
+	if ( 0 < segment.configuration?.favorite_categories?.length ) {
+		getFavoriteCategoryNames(
```

Comment:
Here and in `SegmentGroup`, this function runs on each render, resulting in a ton of unnecessary requests. These calls should be wrapped in `useEffect` hooks.

Reply:
Good call—updated in 549fc70.